test/jdk/javax/sound/SoundClip/SoundClipTest.java

```diff
@@ -0,0 +1,109 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.File;
+import javax.sound.SoundClip;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.Mixer;
+
+/*
+ * @test
+ * @bug 8356049
+ * @key sound headful
+ * @summary basic testing of javax.sound.SoundClip
+ * @run main/othervm SoundClipTest javasound.wav
+ */
+
+public class SoundClipTest {
+
+    public static void main(String[] args) throws Exception {
+
+        if (!isSoundcardInstalled()) {
+            return;
+        }
+        String dir = System.getProperty("test.src", ".");
+        String sep = System.getProperty("file.separator");
+        File file = new File(dir + sep + args[0]);
+
+        SoundClip clip = SoundClip.createSoundClip(file);
+
+        if (!clip.canPlay()) {
+            throw new RuntimeException("Cannot play clip");
+        }
+        boolean playing = false;
+        int waitCount = 0;
+        clip.loop();
+        while (!playing && waitCount < 10) {
+            Thread.sleep(500);
+            if (clip.isPlaying()) {
+                playing = true;
+                break;
+            }
+            waitCount++;
+        }
+        if (!playing) {
+            throw new RuntimeException("Clip does not play");
+        }
+        waitCount = 0;
+        clip.stop();
+
+        while (playing && waitCount < 10) {
+            Thread.sleep(500);
+            if (clip.isPlaying()) {
+                playing = false;
+                break;
+            }
+            waitCount++;
+        }
+        if (!playing) {
+            throw new RuntimeException("Clip does not stop");
+        }
+
+        // Should also test play() but don't check isPlaying() for test reliability reasons.
+        clip.play();
+        Thread.sleep(1000);
+        clip.stop();
+    }
+
+   /**
+     * Returns true if at least one soundcard is correctly installed
+     * on the system.
+     */
+    public static boolean isSoundcardInstalled() {
+        boolean result = false;
+        try {
+            Mixer.Info[] mixers = AudioSystem.getMixerInfo();
+            if (mixers.length > 0) {
+                result = AudioSystem.getSourceDataLine(null) != null;
+            }
+        } catch (Exception e) {
+            System.err.println("Exception occurred: "+e);
+        }
```

Comment:
The stack trace of the exception could be crucial to understanding what went wrong, so printing the stack trace could be a better approach than just the error message.

I guess you want to pass the test (or as I suggest throw `jtreg.SkippedException`) if anything goes wrong.

`AudioSystem.getSourceDataLine` throws `LineUnavailableException` which is checked exception, and if the exception

Reply:
I think I'd like to get to a place where  if you pass @sound these tests should skip this testing and actually FAIL if they can't find sound. Only allowing the skipping if you didn't explicitly pass @sound. 

I haven't found a way to know if that's possible. I'm going to ask ..

This method is copied straight from another test. I think lots of tests use it.
I guess since it already prints the exception adding a trace isn't a problem.

But it isn't completely reliable. Some systems (cloud ones with "emulated" sound) pass but then the test hangs. I'm probably going to have to do some test exclusion if I can't figure that out but I don't expect it to be in this PR because it'll be something internal to our CI systems.